src/lib/storage/vector_compression/fixed_width_integer/fixed_width_integer_utils.hpp

```diff
@@ -0,0 +1,34 @@
+#pragma once
+
+#include "storage/vector_compression/compressed_vector_type.hpp"
+#include "utils/assert.hpp"
+
+namespace opossum {
+
+inline bool is_fixed_width_integer(CompressedVectorType type) {
+  switch (type) {
+    case CompressedVectorType::FixedWidthInteger4Byte:
+    case CompressedVectorType::FixedWidthInteger2Byte:
+    case CompressedVectorType::FixedWidthInteger1Byte:
+      return true;
+    default:
```

Comment:
In a switch-case statement on enumerations, it can be dangerous to have a default fallback. Without the default clause, the compiler warns you when values are added to the enumeration. With the default clause, the compiler will not report a warning when additional values are added. As finding potential errors during compile time is better than finding them during runtime, I'd add the missing `BitPacking` case and remove the default clause.

Reply:
good point, thank you!

Reply:
but then i need to add an additional return in the end to avoid

> In file included from /var/lib/jenkins/workspace/hyrise_hyrise_PR-2350/src/lib/storage/index/group_key/composite_group_key_index.cpp:16,
                 from /var/lib/jenkins/workspace/hyrise_hyrise_PR-2350/gcc-debug/src/lib/CMakeFiles/hyrise_impl.dir/Unity/unity_26_cxx.cxx:5:
/var/lib/jenkins/workspace/hyrise_hyrise_PR-2350/src/lib/storage/vector_compression/fixed_width_integer/fixed_width_integer_utils.hpp: In function 'bool opossum::is_fixed_width_integer(opossum::CompressedVectorType)':
/var/lib/jenkins/workspace/hyrise_hyrise_PR-2350/src/lib/storage/vector_compression/fixed_width_integer/fixed_width_integer_utils.hpp:17:1: error: control reaches end of non-void function [-Werror=return-type]
   17 | }
      | ^

Reply:
do you think it's still worth it?

Reply:
Pinging @mweisgut .

Reply:
Sorry for the delay. What compiler do you use @benrobby? In some cases, GCC evaluates an unreachable program path as reachable, see https://github.com/hyrise/hyrise/blob/91dcdfd6d86f92d6e8f640893334bbb343ca6fca/src/lib/optimizer/strategy/predicate_placement_rule.cpp#L348-L350 or https://github.com/hyrise/hyrise/blob/91dcdfd6d86f92d6e8f640893334bbb343ca6fca/src/lib/storage/index/abstract_index.cpp#L26-L28).

Reply:
Since we handle all four possible cases, we should fail loudly at this code location instead of returning a default value.